docs/tutorial.rst

```diff
@@ -1092,37 +1126,32 @@ We'll count up the number of neutral and deleterious mutations in the three regi
 .. code-block:: python
 
     num_neutral = np.zeros(3, dtype="int")
-    num_del = np.zeros(3, dtype="int")
+    num_deleterious = np.zeros(3, dtype="int")
     for site in ts.sites():
-        j = int(site.position >= gene_interval[0, 0]) + int(
-            site.position >= gene_interval[0, 1]
-        )
-        unique_muts = {}
+        region = np.digitize(site.position, gene_interval.flatten())
         for mut in site.mutations:
-            for ds, md in zip(mut.derived_state.split(","), mut.metadata["mutation_list"]):
-                if ds not in unique_muts:
-                    unique_muts[ds] = md["selection_coeff"]
+            for mut_info in mut.metadata["mutation_list"]:
+                if mut_info["selection_coeff"] == 0:
+                    num_neutral[region] += 1
```

Comment:
This is not quite right because a neutral mutation that has another mutation stacked on top of it later will be counted twice (that's the point of the `unique_muts` thing). Instead, my proposal would be to do like
```
mut_s = sum([mut_info["selection_coeff"] for mut_info in mut.metadata["mutation_list"]])
if mut_s == 0:
    num_neutral[region] += 1
```
Oh, wait, this doesn't work either, since a neutral mutation placed on top of a non-neutral one wouldn't be counted. Hm...

Reply:
yeah, you're totally right -- the previous version of this code block collapses SLiM mutation ids that occur across multiple `tskit.Mutation`s. So, we'll need to use/explain the `keep_mutation_ids_as_alleles` argument. It'd be great to end up with something a bit less cumbersome than what was in the previous version.

Reply:
hm, I probably borked a test back in #1356. It's still valid -- just checking for the presence of non-neutral mutations doesn't require that they're correctly collapsed -- but probably not good to have a misleading example floating around in the tests.

Reply:
How about
```
mut_s = sum([mut_info["selection_coeff"] for mut_info in mut.metadata["mutation_list"]])
if mut.parent != tskit.NULL:
    parent_mut = ts.mutation(mut.parent)
    parent_s = sum([mut_info["selection_coeff"] for mut_info in parent_mut.metadata["mutation_list"]])
    mut_s -= parent_s
if mut_s == 0:
    num_neutral[region] += 1
```
So, if the net selection coefficient of a mutation is the same as it's parent mutation (if there is one), then it's neutral. This will work in all cases except weird ones that couldn't happen in a stdpopsim simulation.

Reply:
This seems fine -- but I pushed an example using `keep_mutation_ids_as_alleles` a short while ago that gives the right answer when mutations are stacked, and may be a bit clearer. Do you have a preference? (either seem fine to me)

Reply:
Ah, sorry, saw it now. I vote to not put `keep_mutation_ids_as_alleles` in the docs, since people will do it, and then get bad VCFs. This solution works and is not too bad, I think?

Reply:
sure, sounds good to me.

Reply:
After chatting, we think it'd be a good idea to functionalize this and put it in `ext/selection.py` (i.e. a function that takes a `Mutation` and a `TreeSequence` and returns the selection coefficient).